File: mobile/sistem_pos/lib/models/store_profile.dart
class StoreProfile {
  const StoreProfile({
    required this.storeName,
    this.address = '',
    this.phone = '',
    this.footer = 'TERIMA KASIH ~ SILAHKAN DATANG KEMBALI',
  });

  static const defaultProfile = StoreProfile(storeName: 'SISTEM POS');

  final String storeName;
  final String address;
  final String phone;
  final String footer;

  factory StoreProfile.fromJson(Map<String, dynamic> json) => StoreProfile(
        storeName: (json['store_name'] as String?) ?? 'SISTEM POS',
        address: (json['address'] as String?) ?? '',
        phone: (json['phone'] as String?) ?? '',
        footer: (json['footer'] as String?) ??
            'TERIMA KASIH ~ SILAHKAN DATANG KEMBALI',
      );

  Map<String, dynamic> toJson() => {
        'store_name': storeName,
        'address': address,
        'phone': phone,
        'footer': footer,
      };
}

File: backend/app/models/store_profile.py
from datetime import datetime

from sqlalchemy import DateTime, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class StoreProfile(Base):
    __tablename__ = "store_profiles"

    id: Mapped[int] = mapped_column(primary_key=True)
    store_name: Mapped[str] = mapped_column(
        String(100), default="SISTEM POS", server_default="SISTEM POS"
    )
    address: Mapped[str | None] = mapped_column(String(255), nullable=True)
    phone: Mapped[str | None] = mapped_column(String(30), nullable=True)
    footer: Mapped[str] = mapped_column(
        String(120),
        default="TERIMA KASIH ~ SILAHKAN DATANG KEMBALI",
        server_default="TERIMA KASIH ~ SILAHKAN DATANG KEMBALI",
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
    )
